go/proc/counting_listener.go

```diff
@@ -37,8 +38,8 @@ type countingConnection struct {
 func Published(l net.Listener, countTag, acceptTag string) net.Listener {
 	return &CountingListener{
 		Listener:   l,
-		ConnCount:  stats.NewInt(countTag),
-		ConnAccept: stats.NewInt(acceptTag),
+		ConnCount:  stats.NewGauge(countTag, "Active connections accepted by counting listener"),
```

Comment:
It has always been confusing to me that `ConnCount` and `ConnAccepted` don't have more descriptive names, and these generic descriptions don't help either :)

While I'm happy to defer this to a subsequent improvement, IMO we should change the `proc.Listen` API to include a description string and pass that here as well along with the countTag.

Then these would be:
```
ConnCount: stats.NewGauge(countTag, fmt.Sprintf("Active %s connections", description)),
ConnAccept: stats.NewCounter(countTag, fmt.Sprintf("Accepted %s connections", description)),
```

Since these are only used for HTTP, we would then change `servenv.Run` to include "HTTP" as the description string.

I also think we would be well served to rename the generic metric names as well to be `HTTPConnCount` and `HTTPConnAccepted` but that would break backwards compatibility. @sougou / @michael-berlin thoughts on this?

Reply:
Agreed, but let's do it in a follow-on?

Reply:
Yup. Would still like to hear from @sougou and @michael-berlin about the backwards-incompatibility of this proposed change.